### PR TITLE
EDU-1235 consolidate room media storage

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-1235

### CODE REVIEW

File: src/server/room-controller.js [hunk 1]
```diff
@@ -447,6 +499,7 @@ export default class RoomController {
   registerPages(router) {
     router.get(
       '/rooms/:roomId*',
+      needsAuthentication(),
```
Comment (adrianaluchian) [inline src/server/room-controller.js:502]: Good catch! 😳

File: src/services/room-service.js [hunk 2]
```diff
@@ -276,6 +374,59 @@ export default class RoomService {
...
+    });
+
+    await this.roomStore.updateRoomMessages(room._id, messages);
+
+    const updatedRoom = await this.roomStore.getRoomById(room._id);
+
+    return updatedRoom;
+  }
+
+  async updateRoomMetadata(roomId, { name, slug, isCollaborative, description }) {
```
Comment (adrianaluchian) [inline src/services/room-service.js:402]: Just noticing that this method belongs in the `// ROOMS` section of the module.

File: src/server/room-controller.js (hunk 1 above)
Comment (ahelmberger) [reply src/server/room-controller.js:502]: It replaces the deleted `if (!user) { throw new Unauthorized(); }` :-) (so no change in behavior actually)

File: src/services/room-service.js (hunk 2 above)
Comment (ahelmberger) [reply src/services/room-service.js]: True, I moved it (and removed the header comments in this file).

File: src/domain/schemas/room-schemas.js [hunk 3]
```diff
@@ -2,6 +2,15 @@ import joi from 'joi';
 import { ROOM_USER_ROLE } from '../constants.js';
 import { emailSchema, idOrKeySchema, slugSchema } from './shared-schemas.js';
 
+export const getAllOrPostRoomMediaParamsSchema = joi.object({
```
Comment (adrianaluchian) [inline src/domain/schemas/room-schemas.js:5]: Tiny thing, I would vote for having two schemas here, even if they are the same, than one called `AOrB`.
If that's also fine with you.

Comment (ahelmberger) [reply src/domain/schemas/room-schemas.js]: Totally fine, split into two it is :-)

**Comment by adrianaluchian** [comment]
You've done a really thorough cleanup and separation of concerns, congrats on keeping all the balls in the air!
Hard to spot if there are any potential hidden issues in this code, but to me it looked good! 👏

**Comment by ahelmberger** [comment]
> You've done a really thorough cleanup and separation of concerns, congrats on keeping all the balls in the air! Hard to spot if there are any potential hidden issues in this code, but to me it looked good! clap

Thanks, I tested the most important aspects, so I don't think there's gonna be a major problem or showstopper, but we need thorough testing for this by Theresa.

**Pull request merged**